SOC/common/simulation.py

```diff
@@ -118,6 +126,44 @@ def plot_state(self, with_boundaries = False):
         
         plt.colorbar(IM)
         return fig
+
+    def animate_states(self, notebook = False, with_boundaries = False):
+        """
+        Animates the current state of the animation.
+        """
+        fig, ax = plt.subplots()
+
+        if with_boundaries:
+            values = np.dstack(self.saved_snapshots)
+        else:
+            values = np.dstack(self.saved_snapshots)[self.BOUNDARY_SIZE:-self.BOUNDARY_SIZE, self.BOUNDARY_SIZE:-self.BOUNDARY_SIZE, :]
+
+        IM = ax.imshow(values[:, :, 0],
+                       interpolation='nearest',
+                       vmin = values.min(),
+                       vmax = values.max()
+                       )
+        
+        plt.colorbar(IM)
+        iterations = values.shape[2]
+        title = ax.set_title("Iteration {}/{}".format(0, iterations * self.save_every))
+
+        def animate(i):
+            IM.set_data(values[:,:,i])
+            title.set_text("Iteration {}/{}".format(i * self.save_every, iterations * self.save_every))
+            return IM, title
+
+        anim = animation.FuncAnimation(fig,
+                                       animate,
+                                       frames=iterations,
+                                       interval=30,
+                                       )
+        if notebook:
+            from IPython.display import HTML, display
```

Comment:
wow, ale lifehacki :) 
generacja html

Reply:
:) to na tyle popularne narzędzie, zwłaszcza że my będziemy go używać, że uznałem że się przyda